services/src/main/java/com/linkedin/kmf/common/TopicFactory.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.linkedin.kmf.common;
+
+import java.util.Map;
+
+
+/**
+ * Constructs the monitored topic if it does not exist.
+ *
+ * Implementations of this class should have a public, no-arg constructor.
```

Comment:
I agree that if it is a class with util class whose methods are self-contained, then we don't need properties in the constructor. Otherwise, we either need properties in the constructor, or we need a separate init() method in the interface. I think the plugin classes in Kafka also follow this pattern, right?

TopicFactory needs to know the datavault uri in order to create topic. If we don't provide it in TopicFactory constructor or some init() method, then we have to provide it in createTopicIfNotExist(). However, I find it a bit weird for this method to require datavault uri, because datavault uri is actually a config of the factory class instead of per-topic config. What do you think?